openwisp_monitoring/device/admin.py

```diff
@@ -63,10 +63,32 @@ class CheckInline(GenericStackedInline):
     fields = ['check_type', 'is_active']
     readonly_fields = ['check_type']
 
+    def _set_check_inline(self):
+        self.fields = []
+        self.readonly_fields = ['created', 'modified']
+
     def has_add_permission(self, request, obj=None):
+        if request.user.has_perm(f'{self.model._meta.app_label}.add_check_inline'):
+            self._set_check_inline()
+            return True
+        return False
+
+    def has_view_permission(self, request, obj=None):
+        if request.user.has_perm(f'{self.model._meta.app_label}.view_check_inline'):
+            self._set_check_inline()
+            return True
+        return super(GenericStackedInline, self).has_view_permission(request, obj)
+
+    def has_change_permission(self, request, obj=None):
+        if request.user.has_perm(f'{self.model._meta.app_label}.change_check_inline'):
+            self._set_check_inline()
+            return True
         return False
```

Comment:
The current code won't allow a user that has permission to edit the `Check` model to edit the inline form. @nemesisdesign is this a desired quality?

Reply:
I have updated this :+1:

Reply:
so it sounds like this was not intentional, right @Aryamanz29?

Reply:
@nemesisdesign https://github.com/openwisp/openwisp-monitoring/pull/431#discussion_r950688809